src/main/java/org/gridsuite/modification/modifications/TwoWindingsTransformerModification.java

```diff
@@ -504,22 +503,24 @@ private void processRegulating(RatioTapChangerModificationInfos ratioTapChangerI
                 AttributeModification.toAttributeModification(0d, OperationType.SET), TARGET_DEADBAND, 2);
             ratioTapChangerReports.add(targetDeadbandReportNode);
         }
-
-        RatioTapChanger.RegulationMode regulationMode = Boolean.TRUE.equals(regulatingValue) ? RatioTapChanger.RegulationMode.VOLTAGE : RatioTapChanger.RegulationMode.REACTIVE_POWER;
-        AttributeModification<RatioTapChanger.RegulationMode> regulationModeModification = AttributeModification.toAttributeModification(regulationMode, OperationType.SET);
-        ReportNode regulationModeReport = ModificationUtils.getInstance().applyElementaryModificationsAndReturnReport(
-            isModification ? ratioTapChanger::setRegulationMode : ratioTapChangerAdder::setRegulationMode,
-            isModification ? ratioTapChanger::getRegulationMode : () -> null,
-            regulationModeModification, "Voltage regulation mode set to " + regulationMode, 1);
-        if (regulationModeReport != null) {
-            ratioTapChangerReports.add(regulationModeReport);
-        }
-        if (regulationModeReport != null) {
+        Boolean isRegulating = ratioTapChangerInfos.getRegulating() != null ? ratioTapChangerInfos.getRegulating().getValue() : null;
```

Comment:
Why do we have isRegulating and regulationMode actually ?

Reply:
Yes but in the front and in the DTO, we should have only the regulation mode. And deduct isRegulating from the regulation mode at the modification application. It makes the code complicated for nothing. Anyway, we won't fix it here